app/src/components/buttons/filterButton/filterContent/filterInput/filterInput.jsx

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright 2024 EPAM Systems
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import classNames from 'classnames/bind';
+import { Dropdown, FieldText } from '@reportportal/ui-kit';
+import PropTypes from 'prop-types';
+import styles from './filterInput.scss';
+
+const cx = classNames.bind(styles);
+
+export const FilterInput = ({ filter, setFilters }) => {
+  const { filterName, options, value, condition, placeholder, title, withField } = filter;
+
+  return (
+    <div className={cx('filter-item')}>
+      <span className={cx('label')}>{title}</span>
+      <div className={cx('container')}>
+        <Dropdown
+          options={options}
+          value={withField ? condition : value}
+          onChange={(newValue) => {
+            setFilters((prevFilters) => ({
```

Comment:
Can we just set the changed value in this component?
Without relying on the logic on how they are stored and without duplicating `...prevFilters,` every time.

Reply:
With this approach, you will have to create a state for each filter field.

Reply:
I mean that this component should just call the callback come from parent with the value.
It is not required here to know how to set the new value and that `prevFilters` should be copied to state every time.

Reply:
I don't understand